### PR TITLE
ci: auto-approve dependabot PRs

### PR DESCRIPTION
**Pull request by flub**
To smoothen the dependabot workflow a bit let's try out auto-approving
dependabot PRs.  This means we should be able to merge these PRs
without having to do the, checks notes, 4 clicks this takes.  Already
a massive improvement.

#skip-changelog

### CODE REVIEW

**Review by loewenheim — APPROVED** [review]
Looks sensible to me, FWIW. I don't have a lot of experience with these CI files.

**Comment by relaxolotl** [comment]
To be clear, what's the workflow after this is merged in? Does it still require some kind of manual action from a developer to finalize the change, or does this automatically approve and merge in dependabot PRs?

**Comment by flub** [comment]
Yep, this does not mean PRs will be auto-merged!  We'll still have to merge this with an explicit action, but we won't have to go do the approval stuff.  I hope, that together with #714 - which still needs updating - I can get merging of dependabot PRs down to creating one comment containing
```
@dependabot merge
#skip-changelog
```
At least in the best case when they can be merged without any further work.

**Pull request merged**
